moshi/src/main/java/com/squareup/moshi/JsonReader.java

```diff
@@ -569,6 +573,32 @@ public final String getPath() {
     return JsonScope.getPath(stackSize, scopes, pathNames, pathIndices);
   }
 
+  /** Returns the tag value for the given class key. */
+  @CheckReturnValue
+  public final @Nullable Object getTag(Class<?> clazz) {
+    if (tags == null) {
+      return null;
+    }
+    return tags.get(clazz);
+  }
+
+  /** Returns all tag values as an unmodifiable list. */
+  @CheckReturnValue
+  public final Map<Class<?>, Object> getTags() {
```

Comment:
Drop this method please, if we expose it then adding a tag becomes an implicit contract that other code can rely upon.

Reply:
removed.